Greatly extend the functionality and scope of this script

johnrigler opened the pull request:
I have been using this version with dogecoin rather than bitcoin because I create bundles of many transactions to build out a communication bus and the cost would be prohibitive with bitcoin.  THIS IS A MASSIVE HACK BUT IT WORKS SO WELL.  I have added many commented into the base code as well as some functionality that supports the dimecash project: https://dime.cash/doge

Namely, small letters and spaces are converted to capitals, numbers are shifted into a special octal representation, most other ascii characters are added, and a "key" has been created that facilitates the support of all 58 initial characters and thus portability to the altcoins.

To use this with zcash, the code would need to be changed to allow for one more character in the string, but I tested that out a few months ago and it works fine.

Even though you pass a "D" as the first argument of this version, it is hard-coded to start the string with DCx which (along with zzz as the final characters) makes the address easy for me to catch via the dimecash site (still a work in progress)

johnrigler committed:
Changed to be more generic

johnrigler committed:
changed spacing

johnrigler committed:
unpack.bash

johnrigler committed:
unpack.bash

johnrigler committed:
remove cruff

johnrigler committed:
Revert "Changed to be more generic"

This reverts commit 446031a9e564af1023f05248d63d08b631aa5140.

johnrigler committed:
Special just for Dogecoin
Works from A to T as first character

johnrigler committed:
new file:   .circleci/config.yml
	modified:   unspendable.py

johnrigler committed:
added repo and pip requirements.txt

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
move hello

johnrigler committed:
.

johnrigler committed:
i

johnrigler committed:
x

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
i

johnrigler committed:
x

johnrigler committed:
.

johnrigler committed:
u

johnrigler committed:
i

johnrigler committed:
i

johnrigler committed:
added test

johnrigler committed:
i

johnrigler committed:
l

johnrigler committed:
replace git pull with checkout

johnrigler committed:
x

johnrigler committed:
x

johnrigler committed:
x

johnrigler committed:
test

johnrigler committed:
changed unpack function

johnrigler committed:
Dockerize and harden CI Pipeline

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
Jenkins test

johnrigler committed:
.

johnrigler committed:
test python2 with jenkins

johnrigler committed:
Create Jenkinsfile

johnrigler committed:
.

johnrigler committed:
Merge branch 'master' of https://github.com/johnrigler/unspendable

Commit:
.

johnrigler committed:
this stuff is cruff

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
DCx,DCy,etc now work as prefix

johnrigler committed:
Merge branch 'cruff'

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
Returned to the original unspendable.py fork
because I broke something.

johnrigler committed:
Branched the last working verison to v1.0

johnrigler committed:
added back circleci with a list of unspendable addresses to test for backwards compatibility
I created many of these by padding outside of the python script

johnrigler committed:
updated readme

johnrigler committed:
work on this in v2.0

johnrigler committed:
added a test directory for circleci

johnrigler committed:
Break test to verify circleci

johnrigler committed:
fix job to verify circleci

johnrigler committed:
change padding to small z

johnrigler committed:
updated v1.0

johnrigler committed:
make test understand environment

johnrigler committed:
add test to v1.0

johnrigler committed:
bug in test

johnrigler committed:
fix bug in test

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
Merge branch 'v1.0'

johnrigler committed:
Merge pull request #1 from johnrigler/v1.0

V1.0

johnrigler committed:
Merge branch 'master' of github.com:johnrigler/unspendable

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
Pulled over code from v1.0

johnrigler committed:
.

johnrigler committed:
new version seems backwards compatible

johnrigler committed:
updated README.md to include v2 stuff

johnrigler committed:
changed simple sum

johnrigler committed:
.

johnrigler committed:
transpose b and d meaning

johnrigler committed:
Updated the X field to support DASH

johnrigler committed:
Removed shocktal number change

johnrigler committed:
Merge branch 'master' of github.com:johnrigler/unspendable

johnrigler committed:
small h now stands for period

johnrigler committed:
new file:   Jenkinsfile

johnrigler committed:
.

johnrigler committed:
q

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
Added Jenkins for Jira test

johnrigler committed:
.

johnrigler committed:
.

johnrigler committed:
Merge branch 'master' of github.com:johnrigler/unspendable

johnrigler committed:
fixed capital I issue

johnrigler committed:
.

johnrigler committed:
Fixed I issue

johnrigler committed:
Added echoland functionality for daily broadcast

johnrigler committed:
Adding initial casts to the system.

johnrigler committed:
Fix issue with 0

johnrigler committed:
modified:   LICENSE
	modified:   README.md
	new file:   un3.cgi
	new file:   unspendable.html

johnrigler committed:
.

Commit:
Big change, transposed period and comma.

johnrigler committed:
Update README.md

CarolineFowler committed:
Create macdougal.js

Draws out a mapping of the MacDougal Base58 text system.

CarolineFowler committed:
Update README.md

CarolineFowler committed:
Update README.md

johnrigler committed:
Merge pull request #2 from CarolineFowler/master

Documentation Update

johnrigler committed:
Update README.md

Added explanation of coded preface.

adamkrellenstein commented:
Hey, just came across this again! I think this is a great project to dramatically improve the original, very modest script I wrote. Since it's so much bigger than the original, do you think maybe it'd be better to just create your own repo and then we can link to it in the README of this one?

johnrigler commented:
Hi Adam. Thanks. I am not great at email, so sorry if there is a delay 
in replying to you.
I could make it into its own repo, but I am not sure what that would 
buy. Also, I really hacked
into it and think there may be some bugs. I can't create any address for 
some reason:

***@***.***:~$ un DCx adfsadf
DCxADFSADFzzzzzzzzzzzzzzzzzzWV3m2P
***@***.***:~$ un Da zzzzz
DBDxaTGiHpY8BZruyafgWsjDNVk3j268Ke
***@***.***:~$ un DZ zzzzzz
DADxaTGHHpY8BZruyafgWsjDNVk3iwXcZM

I am not sure if you have any cycles to look at this, but then if you 
make an update, I could inherit it. Either way is cool. I am starting to 
drift away from github now and just putting stuff directly into a mix of 
IPFS and EVM transactions.


On 2023-10-27 03:46, Adam Krellenstein wrote:
> Hey, just came across this again! I think this is a great project to
> dramatically improve the original, very modest script I wrote. Since
> it's so much bigger than the original, do you think maybe it'd be
> better to just create your own repo and then we can link to it in the
> README of this one?
> 
> --
> Reply to this email directly, view it on GitHub [1], or unsubscribe
> [2].
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> 
> Links:
> ------
> [1] 
> https://github.com/adamkrellenstein/unspendable/pull/2#issuecomment-1782536945
> [2] 
> https://github.com/notifications/unsubscribe-auth/ABXXPHKLIAKLE3MUKG5IJU3YBNYHVAVCNFSM4IYYFDW2U5DIOJSWCZC7NNSXTN2JONZXKZKDN5WW2ZLOOQ5TCNZYGI2TGNRZGQ2Q

adamkrellenstein commented:
ach I'm sorry, I won't have time to look at this. best of luck though!

johnrigler commented:
On 2023-11-01 05:08, Adam Krellenstein wrote:

No worries. I think I will take you advice and spin up a new version at 
some
point that points back to yours, so if you put a link in your README, it
would alert more people about the extended project.

Hope all is well. Cheers.

> ach I'm sorry, I won't have time to look at this. best of luck though!
> 
> 
> --
> Reply to this email directly, view it on GitHub [1], or unsubscribe
> [2].
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> 
> Links:
> ------
> [1] 
> https://github.com/adamkrellenstein/unspendable/pull/2#issuecomment-1788707259
> [2] 
> https://github.com/notifications/unsubscribe-auth/ABXXPHOKDCO6KSGJV4YDW2LYCINTNAVCNFSM4IYYFDW2U5DIOJSWCZC7NNSXTN2JONZXKZKDN5WW2ZLOOQ5TCNZYHA3TANZSGU4Q

johnrigler committed:
Add files via upload

This is a fully self-contained vanilla Javascript version that supports the following ranges of Digibyte transactions:

D5	DB	DH	DP	SM	ST	SZ	Sf
D6	DC	DJ	DQ	SN	SU	Sa	Sg
D7	DD	DK	DR	SP	SV	Sb	Sh
D8	DE	DL	DS	SQ	SW	Sc	Si
D9	DF	DM	DT	SR	SX	Sd	Sj
DA	DG	DN	DU	SS	SY	Se	Sk

johnrigler committed:
Update and rename un4a.html to unspendable_js.html
